z/buffer.go

```diff
@@ -57,7 +57,7 @@ type Buffer struct {
 }
 
 func NewBuffer(capacity int, tag string) *Buffer {
-	if capacity == 0 {
+	if capacity < defaultCapacity {
```

Comment:
I think it should be okay to increase the capacity upto at least `defaultCapacity=64`. 
Do let me know if it is better to add an extra check like:
`capacity = max(capacity, 8) // 8 is default padding`